### PR TITLE
Export for OS X on OS X now lets you select .dmg or .zip

### PR DESCRIPTION
**Pull request by marcelofg55**

### CODE REVIEW

File: editor/editor_export.cpp [hunk 1]
```diff
@@ -1351,18 +1351,21 @@ bool EditorExportPlatformPC::can_export(const Ref<EditorExportPreset> &p_preset,
 	return valid;
 }
 
-String EditorExportPlatformPC::get_binary_extension(const Ref<EditorExportPreset> &p_preset) const {
+List<String> EditorExportPlatformPC::get_binary_extension(const Ref<EditorExportPreset> &p_preset) const {
```
Comment (akien-mga) [inline editor/editor_export.cpp]: Should probably be "get_binary_extension**s**" then.

File: editor/editor_export.cpp [hunk 2]
```diff
@@ -1351,18 +1351,21 @@ bool EditorExportPlatformPC::can_export(const Ref<EditorExportPreset> &p_preset,
...
+	List<String> list;
 	for (Map<String, String>::Element *E = extensions.front(); E; E = E->next()) {
 		if (p_preset->get(E->key())) {
-			return extensions[E->key()];
+			list.push_back(extensions[E->key()]);
+			return list;
```
Comment (akien-mga) [inline editor/editor_export.cpp]: Should it return as soon as it finds one, or finish populating `list` with valid extensions and only return at the end?

File: editor/editor_export.cpp (hunk 1 above)
Comment (marcelofg55) [reply editor/editor_export.cpp]: Yeah, makes sense, I'll change that.

File: editor/editor_export.cpp (hunk 2 above)
Comment (marcelofg55) [reply editor/editor_export.cpp]: I returned the list immediately to keep the logic of the function as it used to be. But I guess it should finish populating the list, I'll try it out.

Comment (marcelofg55) [reply editor/editor_export.cpp]: This only affects the Linux export, if I let it finish populating the list:
screenshot from 2018-10-31 16-00-17

Sounds okay?

Comment (akien-mga) [reply editor/editor_export.cpp]: Seems fine, though normally `x86_64` is for 64-bit and `x86` is for 32-bit -- but does it mean that prior to this change, 32-bit export would also propose only `x86`?
And now, what's the default extension when doing a 32-bit export for Linux?

Comment (marcelofg55) [reply editor/editor_export.cpp]: So I've just checked, if I keep the functionality as it was, returning the list immediately, the linux export gets .x86 when the export is 32-bit and .x86_64 when the export is 64-bit. Seems better this way, I'll keep functionality as it is then.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**